rotkehlchen/chain/evm/decoding/merkl/utils.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from typing import Any
+
+from rotkehlchen.errors.misc import RemoteError, UnableToDecryptRemoteData
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChainID
+from rotkehlchen.utils.network import request_get
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def _paginated_request(url_template: str, items_per_page: int, context: str, **url_params: Any) -> list | None:  # noqa: E501
+    """Make paginated requests to Merkl API.
+
+    `url_template` is the URL template with {page} and {items} placeholders
+    `items_per_page` is the number of items per page
+    `context` is the context string for error logging
+    `**url_params` are the additional parameters for URL formatting
+
+    Returns a list of all items from all pages, or None if there was an error.
+    """
+    all_items: list[Any] = []
+    page = 0
+    while True:
+        try:
+            url = url_template.format(page=page, items=items_per_page, **url_params)
+            all_items.extend(response := request_get(url))
+            if len(response) == 0 or len(response) < items_per_page:  # If we get less than the full page size, we've reached the end  # noqa: E501
+                break
```

Comment:
```suggestion
            if len(response) < items_per_page:  # If we get less than the full page size, we've reached the end  # noqa: E501
                break
```

as I said the case 0 is covered by the other inequality